template/src/navigation/RouteKey.ts

```diff
@@ -1,19 +1,17 @@
-import {IRouteKey} from '../constants/interface/common/CommonInterface'
-
-const RouteKey: IRouteKey = {
+export enum RouteKey {
   /** Screen */
-  SplashScreen: 'SplashScreen',
-  LoginScreen: 'LoginScreen',
-  SignUpScreen: 'SignUpScreen',
-  HomeScreen: 'HomeScreen',
+  SplashScreen = 'SplashScreen',
+  LoginScreen = 'LoginScreen',
+  SignUpScreen = 'SignUpScreen',
+  HomeScreen = 'HomeScreen',
 
   /** Stack */
-  AuthStack: 'AuthStack',
-  MainStack: 'MainStack',
-  HomeStack: 'HomeStack',
+  AuthStack = 'AuthStack',
+  MainStack = 'MainStack',
+  HomeStack = 'HomeStack',
 
   /** Tab */
-  MainTab: 'MainTab',
+  MainTab = 'MainTab',
 }
 
-export default RouteKey
+export default RouteKey;
```

Comment:
can you extend keys from AppStackParamList?

like this: 
import {AppStackParamList} from './types'

const RouteKey: Record<keyof AppStackParamList, keyof AppStackParamList> = {
  HomeScreen: 'HomeScreen',
  LoginScreen: 'LoginScreen',
  SignUpScreen: 'SignUpScreen',
}

export default RouteKey

Reply:
updated